pennylane_forest/qpu.py

```diff
@@ -83,6 +86,28 @@ def __init__(self, device, *, shots=1024, active_reset=True, load_qc=True, reado
 
         self._eigs = {}
 
+        self._compiled_program = None
+        """Union[None, pyquil.ExecutableDesignator]: the latest compiled program. If parametric
+        compilation is turned on, this will be a parametric program."""
+
+        self.parametric_compilation = kwargs.get("parametric_compilation", True)
+
+        if self.parametric_compilation:
+            self._compiled_program_dict = {}
+            """dict[int, pyquil.ExecutableDesignator]: stores circuit hashes associated
+                with the corresponding compiled programs."""
+
+            self._parameter_map = {}
+            """dict: stores the string of symbolic parameters associated with
+                their numeric values. This map will be used to bind parameters in a parametric
+                program using PyQuil."""
+
+            self._parameter_reference_map = {}
+            """dict: stores the string of symbolic parameters associated with
```

Comment:
What are the types of the dictionary keys/values here?

Reply:
Thanks for getting that! Forgot to update this occurrence, whereas the ``QVMDevice`` counterpart has the docstring straight (one extra reason for having these defined at only one place)

Reply:
(Getting at the ``string`` -> ``str`` part as well from ``QVM``)